raiden/tests/scenarios/mfee3_imbalance_fee.yaml

```diff
@@ -0,0 +1,98 @@
+version: 2
+
+settings:
+  gas_price: "fast"
+  chain: any
+  services:
+    pfs:
+      url: https://pfs-goerli-with-fee.services-dev.raiden.network
+    udc:
+      enable: true
+      token:
+        deposit: true
+
+token:
+  address: "0x62083c80353Df771426D209eF578619EE68D5C7A"
+  balance_min: 1_000_000
+  balance_fund: 1_000_000
+
+nodes:
+  mode: managed
+  count: 3
+  raiden_version: local
+
+  default_options:
+    gas-price: fast
+    environment-type: development
+    routing-mode: pfs
+    pathfinding-max-paths: 5
+    pathfinding-max-fee: 100
+    flat-fee:
+      - "0x62083c80353Df771426D209eF578619EE68D5C7A"
+      - 0
+    proportional-fee:
+      - "0x62083c80353Df771426D209eF578619EE68D5C7A"
+      - 0
+    proportional-imbalance-fee:
+      - "0x62083c80353Df771426D209eF578619EE68D5C7A"
+      - 10_000
+
+## This is the MFEE3 scenario. It creates a network with topology 0 -> 1 -> 2 and checks
+## whether a path is returned. It also checks that correct imbalance fees are deducted or added in order
+## to keep the channel imbalances within the desired limits.
+## For every 10000 TKNs tranferred a fee of 10 TKN is expected.
+
+scenario:
+  serial:
+    tasks:
+      - parallel:
+          name: "Open channels"
+          tasks:
+            - open_channel: {from: 0, to: 1, total_deposit: 100_000}
+            - open_channel: {from: 1, to: 2, total_deposit: 100_000}
+      - serial:
+          name: "Test providing routes"
+          tasks:
+            - wait_blocks: 2
+            # Check that the PFS returns a path from 0 to 3
+            - transfer: {from: 0, to: 2, amount: 50_000, expected_http_status: 200}
+
+            ## Check that the path is indeed the expected one
+            - assert_pfs_history:
+                source: 0
+                request_count: 1
+                target: 2
+                expected_routes:
+                  - [0, 1, 2]
+
+            # Test that correct imbalance fees were used
+            # Imbalance fee is 1%
+            # Fee of fee is 1%
```

Comment:
What is fee of fee? The fees I know are: flat, proportional and proportional imbalance

Reply:
The way I understand it, there's an "extra" fee that calculated based on the fact that the earlier in the chain of mediators you are, the more liquidity you need to put up in order to pay for fees to all mediators throughout the route, and hence should be paid a bigger fee.

Reply:
@palango what's that fee? I did not see something like that when working on the fees in the client.

Reply:
So what Jacob means is that in a chain of mediators with proportional fees, there will be fees of fees. For example 0 want to send 100 T to 3, in a network like 0 -> 1 -> 2 -> 3, where 2 and 3 have 10% proportional fees.

This leads to 2 needing to receive 110, which in turn leads to 1 nedding to receive 12**1**, because it also takes a 10% fee on the amount send to 3.

However, this doesn't happen with imbalance fees and this comment is wrong here.

Reply:
Aha I see. That makes sense.